test/Frontend/invalid-module-alias.swift

```diff
@@ -0,0 +1,25 @@
+// Tests for invalid module alias format and values.
+
+// RUN: not %target-swift-frontend -emit-silgen -parse-as-library %S/Inputs/invalid-module-alias.swift -module-name foo -module-alias foo=bar 2>&1 | %FileCheck -check-prefix=INVALID_MODULE_ALIAS %s
+// INVALID_MODULE_ALIAS: error: invalid module alias "foo"; make sure the alias differs from the module name, module ABI name, module link name, and a standard library name
```

Comment:
Is `a standard library name` possible value only `Swift` in practice or does it concern other core libraries? If it has just one possible value we could pass down the string value of `STDLIB_NAME` in the diagnostic.

Reply:
I meant to make it sound more general, so names like lib- are not passed in either.